elastic-utils-lib/src/cmr/elastic_utils/config.clj

```diff
@@ -4,42 +4,80 @@
    [clojure.data.codec.base64 :as b64]
    [cmr.common.config :as config :refer [defconfig]]))
 
-(declare elastic-host elastic-port set-elastic-port!
-         elastic-admin-token elastic-scroll-timeout elastic-query-timeout
-         elastic-unknown-host-retries elastic-unknown-host-retry-interval-ms
-         elastic-scroll-search-type)
+(declare es-unlimited-page-size)
+(defconfig es-unlimited-page-size
+  "This is the number of items we will request from elastic search at a time when
+  the page size is set to unlimited."
+  {:default 10000
+   :type Long})
+
+(declare es-max-unlimited-hits)
+(defconfig es-max-unlimited-hits
+  "Sets an upper limit in order to get all results from elastic search
+  without paging. This is used by CMR applications to load data into their
+  caches."
+  {:default 200000
+   :type Long})
+
+(declare elastic-host)
 (defconfig elastic-host
   "Elastic host or VIP."
   {:default "localhost"})
 
+(declare elastic-port)
 (defconfig elastic-port
   "Port elastic is listening on."
   {:default 9210 :type Long})
 
+(declare elastic-admin-token)
 (defconfig elastic-admin-token
     "Token used for basic auth authentication with elastic."
     {:default (str "Basic " (b64/encode (.getBytes "echo-elasticsearch")))})
 
+(declare elastic-scroll-timeout)
 (defconfig elastic-scroll-timeout
   "Timeout for ES scrolling"
   {:default "5m"})
 
+(declare elastic-query-timeout)
 (defconfig elastic-query-timeout
   "Timeout for ES queries"
   {:default "170s"})
 
+(declare elastic-unknown-host-retries)
 (defconfig elastic-unknown-host-retries
   "Number of times to retry on ES unknown host error"
   {:default 3 :type Long})
 
+(declare elastic-unknown-host-retry-interval-ms)
 (defconfig elastic-unknown-host-retry-interval-ms
   "Number of milliseconds to retry an ES unknown host error"
   {:default 500 :type Long})
 
+(declare elastic-scroll-search-type)
 (defconfig elastic-scroll-search-type
   "Search type to use with scrolling - either 'scan' or 'query_then_fetch'"
   {:default "query_then_fetch"})
 
+;; note for review, is this next value redundent with
+;; indexer-app/src/cmr/indexer/data/index_set.clj
+;; search-app/src/cmr/search/data/elastic_search_index.clj
+(declare collections-index-alias)
+(defconfig collections-index-alias
```

Comment:
it does look exactly the same as the defconfig in index-set and elastic_search_index. Unsure if it's a value that can be universal, or if the values change per app though

Reply:
the code already had redundant values, they are the same and they will be used by all three blocks of code together, I think these exist this way so that you don't need to share libraries just to get a defconf, I'm just pointing out the oddity in case you want to rename these, I'll add more notes there.